ar4si.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package ar4si
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jws"
+)
+
+type TrustTier int8
+
+const (
+	TrustTierNone            TrustTier = 0
+	TrustTierAffirming       TrustTier = 2
+	TrustTierWarning         TrustTier = 32
+	TrustTierContraindicated TrustTier = 96
+)
+
+var (
+	StatusTierToString = map[TrustTier]string{
+		TrustTierNone:            "none",
+		TrustTierAffirming:       "affirming",
+		TrustTierWarning:         "warning",
+		TrustTierContraindicated: "contraindicated",
+	}
+
+	StringToTrustTier = map[string]TrustTier{
+		"none":            TrustTierNone,
+		"affirming":       TrustTierAffirming,
+		"warning":         TrustTierWarning,
+		"contraindicated": TrustTierContraindicated,
+	}
+)
+
+func (o TrustTier) MarshalJSON() ([]byte, error) {
+	var (
+		s  string
+		ok bool
+	)
+
+	s, ok = StatusTierToString[o]
+	if !ok {
+		return nil, fmt.Errorf("unknown trust tier '%d'", o)
+	}
+
+	return json.Marshal(s)
+}
+
+func (o *TrustTier) UnmarshalJSON(data []byte) error {
+	var s string
+
+	if err := json.Unmarshal(data, &s); err != nil {
+		return fmt.Errorf("unable to decode trust tier '%s': %w", string(data), err)
+	}
+
+	r, ok := StringToTrustTier[s]
+	if !ok {
+		return fmt.Errorf("unknown trust tier '%s'", s)
+	}
+
+	*o = r
+
+	return nil
+}
+
+type TrustVector struct {
+	InstanceIdentity int8 `json:"instance-identity"`
+	Configuration    int8 `json:"configuration"`
+	Executables      int8 `json:"executables"`
+	FileSystem       int8 `json:"file-system"`
+	Hardware         int8 `json:"hardware"`
+	RuntimeOpaque    int8 `json:"runtime-opaque"`
+	StorageOpaque    int8 `json:"storage-opaque"`
+	SourcedData      int8 `json:"sourced-data"`
+}
+
+type AttestationResults struct {
+	Status            *TrustTier   `json:"status"`
+	TrustVector       *TrustVector `json:"trust-vector,omitempty"`
+	RawEvidence       *[]byte      `json:"raw-evidence,omitempty"`
+	Timestamp         *time.Time   `json:"timestamp"`
+	AppraisalPolicyID *string      `json:"appraisal-policy-id"`
```

Comment:
Why is this a `*string` rather than a `string`?

Reply:
done in [2697b9d](https://github.com/veraison/ar4si/pull/4/commits/2697b9db52a0bfe568cfc37af35bd14b72ac7c29)